### PR TITLE
[Bug-Fix] Torch divide_no_nan() produces NaN gradients

### LINKED ISSUE #23063: [Bug] Torch divide_no_nan() produces NaN gradients
The Torch implementation uses ```torch.where(x2 == 0, 0, torch.divide(x1, x2))```. Torch still evaluates the divide branch, so forward values can be zero while backward gradients become NaN for zero denominators. JAX and TensorFlow donot do this.

Repro Script: 
```python
import os
os.environ["KERAS_BACKEND"] = "torch"

import keras
import torch

x1 = torch.tensor([1.0, 2.0], requires_grad=True)
x2 = torch.tensor([0.0, 2.0], requires_grad=True)

y = keras.ops.divide_no_nan(x1, x2)
print("y:", y)

y.sum().backward()
print("x1.grad:", x1.grad)
print("x2.grad:", x2.grad)
print("finite:", torch.isfinite(x1.grad), torch.isfinite(x2.grad))
```

output:
```
y: tensor([0., 1.], grad_fn=<WhereBackward0>)
x1.grad: tensor([   nan, 0.5000])
x2.grad: tensor([    nan, -0.5000])
finite: tensor([False,  True]) tensor([False,  True])
```
Existing Keras tests only check forward values, so they miss this gradient bug.

Enviroment: keras 3.15.0

### PR DESCRIPTION
**Pull request by ChiragSW**
## Description
Fixes: #23063 
Changed keras/src/backend/torch/numpy.py (line 2137) so ```divide_no_nan``` now:
- promotes inputs to the correct result dtype
- converts scalar inputs to tensors
- replaces zero denominators with 1 before division
- still masks output to 0 where denominator is zero
Also added a regression test
## Contributor Agreement

**Please review our [AI-Assisted Contribution Policy](../CONTRIBUTING.md#ai-assisted-contribution-policy) and check all boxes below before submitting your PR for review:**

- [x] I am a human, and not a bot.
- [x] I will be responsible for responding to review comments in a timely manner.
- [x] I will work with the maintainers to push this PR forward until submission.

**Note:** Failing to adhere to this agreement may result in your future PRs no longer being reviewed.

### CODE REVIEW

**Review by hertschuh — APPROVED** [review]
Thanks for the fix!
